### ISSUE TITLE
Assert failure: !CREATE_CHECK_STRING(!"Detected use of a corrupted OBJECTREF. Possible GC hole.")

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#24489]
**Job:**
https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm/job/jitstress/job/arm_cross_checked_windows_nt_corefx_jitstress2_tst/66/

**Failed tests:**
System.Dynamic.Runtime.Tests

**Log:**
```
Starting:    System.Dynamic.Runtime.Tests (parallel test collections = on, max threads = 8)

Assert failure(PID 92 [0x0000005c], Thread: 8200 [0x2008]): !CREATE_CHECK_STRING(!"Detected use of a corrupted OBJECTREF. Possible GC hole.")

<no module>! <no symbol> + 0x0 (0x00000000)
    File: d:\j\workspace\arm_cross_che---b006c545\src\vm\object.cpp Line: 670
    Image: C:\j\workspace\arm_cross_che---6fedd1e9\_\fx\artifacts\bin\testhost\netcoreapp-Windows_NT-Release-arm\dotnet.exe

exit code -1073740286
```

**Comment by echesakovMSFT** [comment]
This is likely a dup of https://github.com/dotnet/coreclr/issues/24278

Assigning to myself to confirm

**Comment by RussKeldorph** [comment]
@echesakovMSFT Could you close this as duplicate if you've confirmed that's the case?

**Comment by echesakovMSFT** [comment]
Confirmed that this is a duplicate of #24278 and will be fixed with #24371
